src/cache.rs

```diff
@@ -46,11 +46,12 @@ impl Cache {
             }
         };
 
-        // Otherwise, fall back to ~/.cache/tealdeer
-        let home_dir = try!(env::home_dir().ok_or(
-            CacheError("Could not determine home directory".into())
-        ));
-        Ok(home_dir.join(".cache").join("tealdeer"))
+        // Otherwise, fall back to $XDG_CACHE_HOME/tealdeer.
+        let xdg_dirs = match BaseDirectories::with_prefix("tealdeer") {
+            Ok(dirs) => dirs,
+            Err(_) => return Err(CacheError("$HOME does not exist or is not defined.".into())),
```

Comment:
The XDG cache dir could also be defined depending on the `XDG_CACHE_HOME` variable or on other fallbacks. Maybe make this more generic?

    return Err(CacheError("Could not determine XDG base directory.".into()))